[dnf5] Repo load comps

lukash opened the pull request:
A non-working draft (I need to figure out how to load the comps for tests). Moves the comps loading in with the rest of the repodata extension loading code. One of the consequences is that newly the comps will also stored as a `.solvx` libsolv cache file and loaded from there if the repomd checksum matches.

Looking for feedback whether this is the desired direction or we for some reason don't want to work with comps the same way as with the other metadata types. (First two commits are cleanups that aren't relevant)

Commit:
libdnf/repo/solv_repo: Fix return code check in load_repo_ext()

Commit:
libdnf/repo: Rename the filenames data type to filelists

This is an inconsistency in repodata extension filelists being referred
to as filenames in libdnf. The file name in repodata has the
"-filelists" suffix, so that's what we should use.

In addition to renaming the type internally, this changes the file name
suffix of the libsolv cache file we store. This shouldn't be an issue,
if there were an old file with the "-filenames" suffix, it simply won't
be read. It will be left unused on the filesystem, and while the size
may not be negligible, it's not worth it to go out of our way to delete
these dead files when dnf is run.

Commit:
Move comps loading to repo

j-mracek commented on libdnf/repo/repo.cpp:
```diff
@@ -181,12 +181,15 @@ void Repo::Impl::load_available_repo(LoadFlags flags) {
+        if (!md_filename.empty()) {
+            solv_repo.load_repo_ext(md_filename, RepodataType::COMPS);
+        } else {
+            logger.debug(fmt::format("no comps available for {}", config.get_id()));
```
Do we need to report it? Comps are not mandatory part and they are often not described in repomd. Comps are only used in Fedora and RHEL repositories. If we will report it, shall we also report missing module.yaml?

lukash replied:
Well it's at debug level (which will not be on by default). I added this one just for consistency, this code needs more refactoring and I'm planning on going over the logging too. A message of some kind is a good idea when loading each of the extra data files, either when it's loading it or when it's not loading it :slightly_smiling_face: (or both if we'd want to be rigorous at the expense of being verbose).

j-mracek replied:
Then we have to logger any missing data type which are allowed to load. But I do afraid that it will represent a long list including module filer, file lists, updateinfo, additional data requested bu user and so on. I thing that the information what kind of optional data are not loaded has a low informative value.

j-mracek replied:
I am sorry I see your point. All other types of metadata have the same logging mechanism. What about to drop logging when some types of data are not available? We can discuss it later on.

Commit:
test/libdnf: Update tests for loading comps

Creates repositories with the comps files in test/data.

Replaces test_dump_and_load with just test_dump, as we really should
only test the dumping part and its output, the loading is tested
already. We should, however, test that the dumped data is also loadable.
In this case, it seems best to just test the dumped data are the same as
those that were loaded, so the test is modified that it checks the
dumped comps.xml is the same as the one from the repo.

This commit doesn't update the contents of
repomd-comps-standard/repodata/comps.xml (hence the test still fails as
of this commit) to highlight the differences between the original
comps.xml and the dumped one.

Commit:
libdnf/comps/group: Dump with indentation and in utf-8

Commit:
test/data/repos-repomd: Udate repomd-comps-standard with dumped xml

Updates the comps.xml with the dumped version so that we can check the
dump against it directly. There are minor differences, mainly the
!DOCTYPE tag is missing.

Commit:
test/libdnf/comps/test_group: Use the test fixture temp dir

lukash commented:
Ready for review now, I've fixed the tests and added some more minor fixes to comps code and tests. @pkratoch not necessarily asking for a review but PTAL :slightly_smiling_face:

Commit:
test/python3/libdnf/comps/test_group: Update comps loading

pkratoch commented:
> Ready for review now, I've fixed the tests and added some more minor fixes to comps code and tests. @pkratoch not necessarily asking for a review but PTAL slightly_smiling_face

LGTM

Conan-Kudo commented on libdnf/repo/solv_repo.cpp:
```diff
@@ -93,12 +94,14 @@ namespace libdnf::repo {
             return "-presto";
         case RepodataType::UPDATEINFO:
             return "-updateinfo";
+        case RepodataType::COMPS:
+            return "-comps";
```
Isn't `-groups` also comps?

lukash replied:
Could you be more specific about your point?

Conan-Kudo replied:
Hmm, I guess I was mistaken. For some reason, I thought `groups.xml` existed as reference to comps groups, but it doesn't.

lukash replied:
Okay, this suffix is also just for the libsolv cache file, so I don't think it would be related...

j-mracek commented:
LGTM
